[All Hosts](shared runtime): Update and move shared runtime articles for PPT support (#2356)
OfficeDev/office-js-docs-pr: docs/excel/run-code-on-document-open.md

---
title: Run code in your Office Add-in when the document opens

> description: 'Run code in your Office Add-in add-in when the document opens.'

Reviewer: ```suggestion
description: 'Learn how to run code in your Office Add-in add-in when the document opens.'
```

ms.date: 12/02/2020
localization_priority: Normal
---

# Run code in your Office Add-in when the document opens

[!include[Shared JavaScript runtime requirements](../includes/shared-runtime-requirements-note.md)]

You can configure your Office Add-in to load and run code as soon as the document is opened. This is useful if you need to register event handlers, pre-load data for the task pane, synchronize UI, or perform other tasks before the add-in is visible.

> [!include[Excel shared runtime note](../includes/note-requires-shared-runtime.md)]

Reviewer: ```suggestion
[!include[Shared runtime note](../includes/note-requires-shared-runtime.md)]
```

## How to run code when the document opens

When your add-in is configured to load on document open, it will run immediately. The `Office.initialize` event handler will be called. Place your startup code in the `Office.initialize` or `Office.onReady` event handler.

The following Excel add-in code shows how to register an event handler for change events from the active worksheet. If you configure your add-in to load on document open, this code will register the event handler when the document is opened. You can handle change events before the task pane is opened.

```JavaScript

> //This is called as soon as the document opens.

Reviewer: ```suggestion
// This is called as soon as the document opens.
```

## How to run code when the document opens

When your add-in is configured to load on document open, it will run immediately. The `Office.initialize` event handler will be called. Place your startup code in the `Office.initialize` or `Office.onReady` event handler.

The following Excel add-in code shows how to register an event handler for change events from the active worksheet. If you configure your add-in to load on document open, this code will register the event handler when the document is opened. You can handle change events before the task pane is opened.

```JavaScript
// This is called as soon as the document opens.

> //Put your startup code here.

Reviewer 2: You didn't change this line, but would be great if we can add a space after the `//`! 
```suggestion
// Put your startup code here.
```

    });
  });
}
```

The following PowerPoint add-in code shows how to register an event handler for selection change events from the PowerPoint document. If you configure your add-in to load on document open, this code will register the event handler when the document is opened. You can handle change events before the task pane is opened.

```JavaScript

> //This is called as soon as the document opens.
> //Put your startup code here.

Reviewer: ```suggestion
// This is called as soon as the document opens.
// Put your startup code here.
```

Office.onReady(info => {
  if (info.host === Office.HostType.PowerPoint) {
    Office.context.document.addHandlerAsync(Office.EventType.DocumentSelectionChanged, onChange);
    console.log("A handler has been registered for the onChanged event.");
  }
});

/**
 * Handle the changed event from the PowerPoint document.
 *
 * @param event The event information from PowerPoint
 */
async function onChange(event) {
  console.log("Change type of event: " + event.type);
}
```

## See also

- [Configure your Office Add-in to use a shared JavaScript runtime](configure-your-add-in-to-use-a-shared-runtime.md)

> - [Share data and events between Excel custom functions and task pane tutorial](../tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md)

Reviewer 2: Would it be relevant to include a link to [Work with Events using the Excel JavaScript API](excel-add-ins-events.md) here? This page: https://docs.microsoft.com/en-us/office/dev/add-ins/excel/excel-add-ins-events

Author: added!
